tensorflow-examples/src/main/java/org/tensorflow/model/examples/cnn/fastrcnn/FasterRcnnInception.java

```diff
@@ -287,35 +287,80 @@ public static void main(String[] params) {
+                                            tf.constant(boxes), colors);
+                                    runner = s.runner();
+                                    s.run(tf.init());
+                                    try (TFloat32 outputBoxedImage = (TFloat32) runner.fetch(drawBoundingBoxes).run().get(0)) {
+                                        //convert the 4D input image to 0.0f - 255.0f
+                                        Mul<TFloat32> mul = tf.math.mul(
+                                                tf.constant(outputBoxedImage),
+                                                tf.constant(255.0f)
+                                        );
+                                        runner = s.runner();
+                                        s.run(tf.init());
+                                        try (TFloat32 mulTensor = (TFloat32) runner.fetch(mul).run().get(0)) {
+                                            //recast and reshape to TUint8 3D tensor
+                                            Cast<TUint8> cast = tf.dtypes.cast(tf.reshape(tf.constant(mulTensor),
+                                                    tf.array(
+                                                            mulTensor.shape().asArray()[1],
+                                                            mulTensor.shape().asArray()[2],
+                                                            mulTensor.shape().asArray()[3]
+                                                    )
+                                            ), TUint8.class);
+                                            runner = s.runner();
+                                            s.run(tf.init());
+                                            try (TUint8 castTensor = (TUint8) runner.fetch(cast).run().get(0)) {
+                                                //Create JPEG from the Tensor with quality of 100%
+                                                EncodeJpeg.Options jpgOptions = EncodeJpeg.quality(100L);
+                                                EncodeJpeg encodeJpeg = tf.image.encodeJpeg(tf.constant(castTensor),
+                                                        jpgOptions);
+                                                runner = s.runner();
+                                                s.run(tf.init());
+                                                try (TString outputImageTensor = (TString) runner.fetch(
+                                                        encodeJpeg.contents()).run().get(0)) {
+                                                    //output the JPEG to file
+                                                    WriteFile writeFile = tf.io.writeFile(tf.constant(outputImagePath),
+                                                            tf.constant(outputImageTensor));
+                                                    runner = s.runner();
+                                                    s.run(tf.init());
```

Comment:
No need to re-run the init.

You could fold all these operations into a single graph and call it once. Pulling things out of the session and poking them back in is probably much slower than just executing a single graph.

Reply:
Latest commit has done that now.